src/components/CalculationFromBase.jsx

```diff
@@ -1,175 +1,207 @@
-    newMonthly[1] = 0;
-    newYearly[1] = 0;
-  } else if (newMonthly[2] * 12 > 18200 && newMonthly[2] * 12 <= 45000) {
-    newMonthly[1] = (0.19 * (parseFloat(newMonthly[2]) * 12 - 18200)) / 12;
-  } else if (newMonthly[2] * 12 > 45000 && newMonthly[2] * 12 <= 120000) {
-    newMonthly[1] =
-      (5092 + 0.325 * (parseFloat(newMonthly[2]) * 12 - 45000)) / 12;
-  } else if (newMonthly[2] * 12 > 120000 && newMonthly[2] * 12 <= 180000) {
-    newMonthly[1] = (
-      (29467 + 0.37 * (parseFloat(newMonthly[2]) * 12 - 120000)) /
-      12
-    ).toFixed();
-  } else if (newMonthly[2] * 12 > 180000) {
-    newMonthly[1] = (
-      (51667 + 0.45 * (parseFloat(newMonthly[2]) * 12 - 180000)) /
-      12
-    ).toFixed();
-  }
-  newHourly[1] = (parseFloat(newMonthly[1]) * 12) / 52 / 5 / 8;
-  newDaily[1] = parseFloat(newHourly[1]) * 8;
-  newWeekly[1] = parseFloat(newHourly[1]) * 8 * 5;
-  newFortnightly[1] = parseFloat(newHourly[1]) * 8 * 5 * 2;
-  newMonthly[1] = (parseFloat(newHourly[1]) * 8 * 5 * 52) / 12;
-  newYearly[1] = parseFloat(newDaily[1]) * 5 * 52 * FactorOfReductionByDaysOff;
-
-  //calculate after-tax income [0]
-  newHourly[0] = parseFloat(newHourly[2]) - parseFloat(newHourly[1]);
-  newDaily[0] = parseFloat(newDaily[2]) - parseFloat(newDaily[1]);
-  newWeekly[0] = parseFloat(newWeekly[2]) - parseFloat(newWeekly[1]);
-  newFortnightly[0] =
-    parseFloat(newFortnightly[2]) - parseFloat(newFortnightly[1]);
-  newMonthly[0] = parseFloat(newMonthly[2]) - parseFloat(newMonthly[1]);
-  newYearly[0] = parseFloat(newYearly[2]) - parseFloat(newYearly[1]);
-
-  if (arrayName) {
-    arrayGroup[arrayName][i] = enteredValue;
-  }
-
-  for (let i = 0; i <= 6; i++) {
-    newHourly[i] = parseFloat(newHourly[i]).toFixed();
-    newDaily[i] = parseFloat(newDaily[i]).toFixed();
-    newWeekly[i] = parseFloat(newWeekly[i]).toFixed();
-    newFortnightly[i] = parseFloat(newFortnightly[i]).toFixed();
-    newMonthly[i] = parseFloat(newMonthly[i]).toFixed();
-    newYearly[i] = parseFloat(newYearly[i]).toFixed();
-  }
-
-  setHourly(newHourly);
-  setDaily(newDaily);
-  setWeekly(newWeekly);
-  setFortnightly(newFortnightly);
-  setMonthly(newMonthly);
-  setYearly(newYearly);
-};
-
-export default calculationFromBase;
+import FYData from "./FYData";
+
+const calculationFromBase = (
+  enteredValue,
+  newSuperRate,
+  newNumDayOff,
+  newYear,
+  hourlyBase,
+  setHourly,
+  setDaily,
+  setWeekly,
+  setFortnightly,
+  setMonthly,
+  setYearly,
+  GST,
+  isContract,
+  arrayName,
+  i
+) => {
+  let newHourly = [];
+  let newDaily = [];
+  let newWeekly = [];
+  let newFortnightly = [];
+  let newMonthly = [];
+  let newYearly = [];
+  let arrayGroup = {
+    newHourly,
+    newDaily,
+    newWeekly,
+    newFortnightly,
+    newMonthly,
+    newYearly,
+  };
+
+  let FactorOfReductionByDaysOff = 0;
+
+  isContract && hourlyBase > 0
+    ? (FactorOfReductionByDaysOff = 1 - newNumDayOff / 260)
+    : (FactorOfReductionByDaysOff = 1);

```

Comment:
use normal if else clause for better readibility, inline ternary operator statement usually is used for assignments e.g. xxx = (condition? 'value when true': 'value when false';
```suggestion
if(isContract && hourlyBase > 0) {
   FactorOfReductionByDaysOff -= newNumDayOff / 260);
} else {
   FactorOfReductionByDaysOff = 1;
}
```